### PR TITLE
feat: enable streaming logs for container

### PR DESCRIPTION
**Pull request by jbrockopp**
Based off of https://github.com/go-vela/community/blob/master/proposals/2021/07-22_log-streaming.md

Related to https://github.com/go-vela/community/issues/367 and https://github.com/go-vela/community/issues/368

This reverts the changes made with https://github.com/go-vela/pkg-executor/pull/177.

See https://github.com/go-vela/worker/pull/232 for more details.

An issue was discovered with the current implementation of streaming logs:

https://github.com/go-vela/community/issues/427

This introduces a [different implementation of streaming logs](https://github.com/go-vela/community/blob/master/proposals/2021/07-22_log-streaming.md#option-1).

The first part of this change is adding a new executor flag:

https://github.com/go-vela/worker/blob/41f276dcd59548901fcae83c588faa850bfef076/executor/flags.go#L45-L51

We currently only allow two values to be set with this flag:

https://github.com/go-vela/worker/blob/41f276dcd59548901fcae83c588faa850bfef076/executor/setup.go#L140-L147

This flag will be used to set a field in the `go-vela/worker/executor/linux` client:

https://github.com/go-vela/worker/blob/41f276dcd59548901fcae83c588faa850bfef076/executor/linux/linux.go#L35

This field in the client will be used in a `switch` statement to control functionality in the `Stream` functions:

https://github.com/go-vela/worker/blob/41f276dcd59548901fcae83c588faa850bfef076/executor/linux/service.go#L227

https://github.com/go-vela/worker/blob/41f276dcd59548901fcae83c588faa850bfef076/executor/linux/step.go#L276

### CODE REVIEW

File: executor/linux/service.go [hunk 1]
```diff
@@ -217,19 +219,124 @@ func (c *client) StreamService(ctx context.Context, ctn *pipeline.Container) err
 	}
 	defer rc.Close()
 
-	// set the timeout to the repo timeout
-	// to ensure the stream is not cut off
-	c.Vela.SetTimeout(time.Minute * time.Duration(c.repo.GetTimeout()))
+	// create new buffer for uploading logs
+	logs := new(bytes.Buffer)
+
+	switch c.logMethod {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/service.go:225]: fixed 👍

File: executor/linux/step.go [hunk 2]
```diff
@@ -266,19 +268,124 @@ func (c *client) StreamStep(ctx context.Context, ctn *pipeline.Container) error
 	}
 	defer rc.Close()
 
-	// set the timeout to the repo timeout
-	// to ensure the stream is not cut off
-	c.Vela.SetTimeout(time.Minute * time.Duration(c.repo.GetTimeout()))
+	// create new buffer for uploading logs
+	logs := new(bytes.Buffer)
+
+	switch c.logMethod {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/step.go:274]: fixed 👍

File: executor/linux/service.go [hunk 3]
```diff
@@ -217,19 +219,124 @@ func (c *client) StreamService(ctx context.Context, ctn *pipeline.Container) err
...
+	case "time-chunks":
+		// create new channel for processing logs
+		done := make(chan bool)
+
+		// nolint: dupl // ignore similar code
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/service.go:230]: fixed 👍

**Review by JordanSussman — APPROVED** [review]

**Review by wass3r — APPROVED** [review]
TY - LGTM

**Review by davidvader — APPROVED** [review]

**Pull request merged**